File: app/src/main/java/com/mirkiewicz/kupmito/AddListDialog.java
package com.mirkiewicz.kupmito;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.Layout;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDialogFragment;

import com.mirkiewicz.kupmito.R;

public class AddListDialog extends AppCompatDialogFragment {

    private EditText et_listname, et_listdesc;
    private AddListDialogListener listener;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.addlist_dialog, null);
        builder.setView(view)
                .setTitle(getString(R.string.list_add))
                .setNegativeButton(getString(R.string.cancel), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                })
                .setPositiveButton(getString(R.string.save), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        String name = et_listname.getText().toString();
                        String description = et_listdesc.getText().toString();
                        listener.applyTexts(name,description);
                    }
                });

        et_listname = view.findViewById(R.id.edit_listname);
        et_listdesc = view.findViewById(R.id.edit_listdesc);




        return builder.create();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        try {
            listener = (AddListDialogListener) context;
        } catch (ClassCastException e) {
            e.printStackTrace();
        }

    }

    public interface AddListDialogListener{
        void applyTexts(String name, String description);
    }

}

File: app/src/main/java/com/mirkiewicz/kupmito/AddListDialog.kt
package com.mirkiewicz.kupmito

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatDialogFragment

class AddListDialog : AppCompatDialogFragment() {

    private var et_listname: EditText? = null
    private var et_listdesc: EditText? = null
    private var listener: AddListDialogListener? = null


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val builder = AlertDialog.Builder(activity)
        val inflater = requireActivity().layoutInflater
        val view = inflater.inflate(R.layout.addlist_dialog, null)

        et_listname = view.findViewById(R.id.edit_listname)
        et_listdesc = view.findViewById(R.id.edit_listdesc)

        builder.setView(view)
                .setNegativeButton(getString(R.string.cancel)) { dialogInterface, i -> }
                .setPositiveButton(getString(R.string.save)) { dialogInterface, i ->

//                    val filtered = ".[]#$"
                    var name = et_listname!!.text.toString().trim()
//                    name = name.filterNot { filtered.indexOf(it) > -1 }

                    var description = et_listdesc!!.text.toString().trim()

                    if (description.length >= 100)
                        description = description.substring(0,100)
                    if (name.length >= 100)
                        name = name.substring(0,100)

                    if (name != "") {
                        listener!!.applyTexts(name, description)
                    } else {
                        val text = getString(R.string.empty_field, "name")
                        Toast.makeText(context, text, Toast.LENGTH_SHORT).show()
                    }
                }

        return builder.create()
    }

    override fun onAttach(context: Context) {

        super.onAttach(context)
        try {
            listener = context as AddListDialogListener
        } catch (e: ClassCastException) {
            e.printStackTrace()
        }
    }

    interface AddListDialogListener {
        fun applyTexts(name: String, description: String)
    }
}

